Source/UnitEx.cpp

```diff
@@ -1194,7 +1194,7 @@ int UnitEx::GetLabCurrentTopic()
 		return HFLNOTINITED;
 	}
 
-	LabData* p = (LabData*)((int)(*unitArray) + (unitID * sizeof(OP2Unit)) + labDataOffset);
+	LabData* p = (LabData*)(reinterpret_cast<std::uintptr_t>(*unitArray) + (unitID * sizeof(OP2Unit)) + labDataOffset);

```

Comment:
Minor point: The `(LabData*)` is also a C-style cast, which should probably be updated.